### DISCUSSION TITLE
esm-node-services: exportDefault is not set for React

### DISCUSSION THREAD

**Discussion by paralin** [esm-dev/esm.sh#716] [General] [answered]
@ije When I run parseCjsExports on "react" it returns `exportDefault: false`, when I would expect `exportDefault: true`. If we don't export "default" from React, when we run esbuild on the following code:

```
export * from "react";
import __cjs_exports$ from "react";
export const { Children, Component, Fragment, Profiler, PureComponent, StrictMode, Suspense, __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED, cloneElement, createContext, createElement, createFactory, createRef, forwardRef, isValidElement, lazy, memo, startTransition, unstable_act, useCallback, useContext, useDebugValue, useDeferredValue, useEffect, useId, useImperativeHandle, useInsertionEffect, useLayoutEffect, useMemo, useReducer, useRef, useState, useSyncExternalStore, useTransition, version } = __cjs_exports$;
```

... the generated code fails with "default" not being defined on the export.

However, if we add the following line (which would be done if exportDefault: true):

```
export { default } from 'react';
```

Everything seems OK.

Shouldn't the result of that script be exportDefault: true?

**Comment by ije** [comment]
cjs always has the _default_ export as esm, unless the `__esModule` exists

```js
const React = require("react")
export default React
export { createElement, ... } = React 
```

**Comment by paralin** [comment]
@ije Ok, thanks. Where is that defined? I'm looking here but don't see it: https://github.com/esm-dev/esm.sh/blob/3324cc705a282b0bf7ae7ebcd15495fed838ace5/server/server_handler.go#L1286C1-L1286C25

**Reply by ije** [reply]
form cjs to esm, the `exports` is treated as `default` of esm

**Reply by paralin** [reply]
Where is this in the esm.sh code? Thanks!

**Reply by ije** [reply, marked as answer]
check https://github.com/esm-dev/esm.sh/blob/main/server/build.go#L255-L262
